web-gateway/src/main/java/controllers/ItemController.java

```diff
@@ -166,65 +167,82 @@ private ItemData fromForm(ItemForm itemForm) {
             }
         });
     }
-
     public CompletionStage<Result> getItem(String itemId) {
         return doGetItem(ctx(), itemId, formFactory.form(BidForm.class));
     }
 
     private CompletionStage<Result> doGetItem(Http.Context ctx, String itemId, Form<BidForm> bidForm) {
         return requireUser(ctx, user -> loadNav(user).thenComposeAsync(nav -> {
+
             UUID itemUuid = UUID.fromString(itemId);
             CompletionStage<Item> itemFuture = itemService.getItem(itemUuid)
                     .handleRequestHeader(authenticate(user)).invoke();
             CompletionStage<PSequence<Bid>> bidHistoryFuture = bidService.getBids(itemUuid)
                     .handleRequestHeader(authenticate(user)).invoke();
-            return itemFuture.thenCombineAsync(bidHistoryFuture, (item, bidHistory) -> {
+            User user1 ;
+            if (nav.getUser().isPresent()) {
+                user1 = nav.getUser().get();
+            }
+            else{
+                return  CompletableFuture.completedFuture(redirect(routes.UserController.createUserForm()));
+            }
+
+                return itemFuture.thenCombineAsync(bidHistoryFuture, (item, bidHistory) -> {
+
+                    if (item.getStatus() == ItemStatus.CREATED && !item.getCreator().equals(user1.getId())) {
+                        return forbidden();
+                    }
+
+                    User seller = null;
+                    Optional<User> winner = Optional.empty();
+
+                    if (item.getCreator().equals(user1.getId())) {
+
+                        seller = user1;
+
+
+                    }
+                    if (item.getAuctionWinner().isPresent() && item.getAuctionWinner().get().equals(user1.getId())) {
+
+                        winner = Optional.of(user1);
 
-                if (item.getStatus() == ItemStatus.CREATED && !item.getCreator().equals(user)) {
-                    return forbidden();
-                }
+                    }
```

Comment:
The logic above isn't quite correct:

What the code _should_ do is look up the user data for the item's seller and winner (if there is a winner) by calling `userService.getUser(item.getCreator())` and `userService.getUser(item.getAuctionWinner().get())` (if `item.getAuctionWinner().isPresent()`).

What this code actually does is set's the seller to the current user, or the winner to the current user, only if the seller or the winner _is_ the current user.

For an example of why that's not right, try creating seller and buyer users, then create an item as seller, and view the item as buyer. You get a `NullPointerException`, because the `seller` is set to `null`.

With this change, the `user1` variable will not be needed.